### ISSUE TITLE
Should Nodeshift return an Object of stuff

### ISSUE THREAD

**Issue by lholmquist** [bucharest-gold/nodeshift#76]
Right now, when nodeshift completes successfully it just returns "done".

I wonder if it should return an object with the things that it created? as one big template file?

It would be nice if it also returned what the pods name is.

Possibly other things?

This sort of ties into the testing framework thingy https://github.com/bucharest-gold/entente/issues/151 and #31

**Comment by lholmquist** [comment]
The answer is yes it should return an object.

**Comment by lance** [comment]
What do you think about keeping the output as-is and writing the output to a file? There is a lot there to clutter up the screen for a typical run. We could add `-` as a final parameter (similar to `tar`) to indicate writing the output to `stdout` - or any other flag.

**Comment by lholmquist** [comment]
Totally. I was planning on just returning an object of stuff for the API.
Not outputting it the console

Sent from my iPhone

On Nov 30, 2017, at 4:51 PM, Lance Ball <notifications@github.com> wrote:

What do you think about keeping the output as-is and writing the output to
a file? There is a lot there to clutter up the screen for a typical run. We
could add - as a final parameter (similar to tar) to indicate writing the
output to stdout - or any other flag.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/bucharest-gold/nodeshift/issues/76#issuecomment-348332456>,
or mute the thread

**Comment by lholmquist** [comment]
since we now have different "goals" that can be run, like resources, build, deploy, what is returned could be different, so i wonder if it makes sense to have the object have properties that correspond to those goals.

    {
      "key": "value",
      ... // some other metadata-y stuff
      "resource": {
        .... // Basically populated with what is in the openshift.json from the resource goal
      },
      "build": {
        .... // build related stuff, like the name of the build, and pod?
      }
    }

or something like that?

basically whatever we would need for that cool testing framework that we are "working" on
